src/gripper_service/src/gripper_service.cpp

```diff
@@ -0,0 +1,107 @@
+/*Copyright 2023 Brookhaven National Laboratory
+BSD 3 Clause License. See LICENSE.txt for details.*/
+#include <gripper_service/gripper_service.hpp>
+
+using namespace std::placeholders;
+
+GripperService::GripperService()
+: node_(std::make_shared<rclcpp::Node>("gripper_server_node")),
+  gripper_(kComPort, kSlaveID)
+{
+  RCLCPP_INFO(node_->get_logger(), "Activate the gripper ...");
+  // Clear the registers
+  gripper_.deactivateGripper();
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  // Activate the gripper
+  gripper_.activateGripper();
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  gripper_.setSpeed(0x0F);
+
+  RCLCPP_INFO(node_->get_logger(), "Activation is successful");
+
+  rclcpp::Service<pdf_beamtime_interfaces::srv::GripperControlMsg>::SharedPtr service =
+    node_->create_service<pdf_beamtime_interfaces::srv::GripperControlMsg>(
+    "gripper_service",
+    std::bind(
+      &GripperService::gripper_controller, this, _1, _2));
+  RCLCPP_INFO(node_->get_logger(), "Ready to receive gripper commands.");
+}
+
+rclcpp::node_interfaces::NodeBaseInterface::SharedPtr GripperService::getNodeBaseInterface()
+// Expose the node base interface so that the node can be added to a component manager.
+{
+  return node_->get_node_base_interface();
+}
+
+void GripperService::gripper_controller(
+  const std::shared_ptr<pdf_beamtime_interfaces::srv::GripperControlMsg::Request> request,
+  std::shared_ptr<pdf_beamtime_interfaces::srv::GripperControlMsg::Response> response)
+{
+  int status = 0;
+  try {
+    // conver the request command string to the mapping enum
+    Gripper_Command gripper_command_enum = gripper_command_map_[request->command];
+
+    switch (gripper_command_enum) {
+      case Gripper_Command::ACTIVE:
+        // Activate the gripper
+        gripper_.deactivateGripper();
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
This would be better as a conditional, so that there isn't a forced sleep that creates an operational bottleneck.